common/parameter.hpp

```diff
@@ -231,6 +231,20 @@ struct set_params_visitor {
         }
         return changed;
     }
+
+    template<typename P, typename F>
+    bool set_asset_param(const P& value, F field) {
+        auto obj = state.*field;
+        bool changed = obj.compare(value);
```

Comment:
`set_asset_param` function has nothing to do with assets. I would suggest calling a template comparison function in the original `set_param` function. To make implementation by default through comparison, and for `breakout_parametrs` redefine it.